### PR TITLE
Deletes accept statement_instance instead of variables

### PR DESCRIPTION
**Pull request by flyingsilverfin**
## What is the goal of this PR?
Implementing the first step of https://github.com/graknlabs/graql/issues/115, we change the grammar to accept `statement_instance` for `delete` clauses, not just variables. As a medium term consequence, we remove the ability to limit, sort or offset delete queries.

## What are the changes implemented in this PR?
* Change `variables` to `statement_instance` in `delete` clauses
* Update tests to reflect not just using variables
* Add new test to ensure query throws if `delete` variable not present in the `match`
* Remove `Filterable` from delete queries

### CODE REVIEW

**Comment by flyingsilverfin** [comment]
@haikalpribadi  ready for review :)

File: java/query/MatchClause.java [hunk 1]
```diff
@@ -116,73 +116,40 @@ public MatchClause(Conjunction<Pattern> pattern) {
...
     /**
-     * @param vars a collection of variables to delete for each result of this match clause
+     * @param delete a collection of statements that indicate properties to delete
      * @return a delete query that will delete the given variables for each result of this match clause
      */
     @CheckReturnValue
-    public final GraqlDelete.Unfiltered delete(LinkedHashSet<Variable> vars) {
-        return new GraqlDelete.Unfiltered(this, vars);
+    public final GraqlDelete delete(Collection<? extends Statement> delete) {
+        return new GraqlDelete(this, new ArrayList<>(delete));
```
Comment (haikalpribadi) [inline java/query/MatchClause.java:152]: variable `delete` should be named `statements`

File: grammar/Graql.g4 [hunk 2]
```diff
@@ -39,7 +39,7 @@ query_undefine      :   UNDEFINE    statement_type+ ;
 query_insert        :   MATCH       pattern+    INSERT  statement_instance+
                     |                           INSERT  statement_instance+  ;
```
Comment (haikalpribadi) [inline grammar/Graql.g4:41]: You should remove the new lines between insert and delete, and also make sure `;` are aligned.

File: java/query/MatchClause.java (hunk 1 above)
Comment (flyingsilverfin) [reply java/query/MatchClause.java:152]: ha! i had this before lol ok will go back

File: java/query/GraqlDelete.java [hunk 3]
```diff
@@ -19,108 +19,77 @@
...
-    @Override
-    public Optional<Long> offset(){
-        return this.offset < 0 ? Optional.empty() : Optional.of(this.offset);
-    }
-
-    @Override
-    public Optional<Long> limit() {
-        return limit < 0 ? Optional.empty() : Optional.of(limit);
+    public Set<Variable> vars() {
+        return statements.stream().flatMap(statement -> statement.variables().stream()).collect(Collectors.toSet());
```
Comment (haikalpribadi) [inline java/query/GraqlDelete.java:76]: I should have caught it during the review, but we shouldn't have kept `vars()` in `GraqlDelete`, @flyingsilverfin . Given that it now takes in "instance statements", it should only be holding onto those, not caring about `vars()` and the equality check should be based on `statements` too.

**Pull request merged**